### ISSUE TITLE
MigPlan can be in "Ready" state with PVs discovered but no action selected after namespace switch

### ISSUE THREAD

**Issue by djwhatle** [fusor/mig-controller#153]
I corrected the `namespaces` field on a MigPlan from a namespace that didn't exist `nginx-example` to one that existed and had an associated PV. The MigPlan controller performed PV discovery and kept the MigPlan in "Ready" state for a few seconds before the "Invalid PV action selected" message showed up.

This is an issue because it allows a MigMigration to start prematurely before a PV action is selected.

**MigPlan with INVALID namespace**

```
apiVersion: v1
items:
- apiVersion: migration.openshift.io/v1alpha1
  kind: MigPlan
  metadata:
    creationTimestamp: 2019-06-10T21:07:42Z
    generation: 1
    labels:
      controller-tools.k8s.io: "1.0"
    name: migplan-sample
    namespace: mig
  spec:
    destMigClusterRef:
      name: migcluster-aws
      namespace: mig
    migStorageRef:
      name: migstorage-sample
      namespace: mig
    namespaces:
    - nginx-example
    srcMigClusterRef:
      name: migcluster-local
      namespace: mig
  status:
    conditions:
    - category: Critical
      lastTransitionTime: 2019-06-10T21:47:51Z
      message: Namespaces [nginx-example] not found on the source cluster.
      reason: NotFound
      status: "True"
      type: NamespaceNotFoundOnSourceCluster
    - category: Required
      lastTransitionTime: 2019-06-10T21:47:56Z
      message: The `persistentVolumes` list has been updated with discovered PVs.
      reason: Done
      status: "True"
      type: PvsDiscovered
```

**MigPlan right after changing spec to be an VALID namespace with PVs**

Notice how the PV discovery is completed but there is no error condition due to an unsupported action.
```
apiVersion: v1
items:
- apiVersion: migration.openshift.io/v1alpha1
  kind: MigPlan
  metadata:
    creationTimestamp: 2019-06-10T21:07:42Z
    generation: 1
    name: migplan-sample
    namespace: mig
    resourceVersion: "23003"
  spec:
    destMigClusterRef:
      name: migcluster-aws
      namespace: mig
    migStorageRef:
      name: migstorage-sample
      namespace: mig
    namespaces:
    - mysql-persistent
    persistentVolumes:
    - name: pv5
      supportedActions:
      - copy
      - move
    srcMigClusterRef:
      name: migcluster-local
      namespace: mig
  status:
    conditions:
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:42Z
      message: The `persistentVolumes` list has been updated with discovered PVs.
      reason: Done
      status: "True"
      type: PvsDiscovered
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:43Z
      message: The storage resources have been created.
      status: "True"
      type: StorageEnsured
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:46Z
      message: The migration registry resources have been created.
      status: "True"
      type: RegistriesEnsured
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:46Z
      message: The migration plan is ready.
      status: "True"
      type: Ready
```

**MigPlan after a few more seconds, error condition for invalid PV action has been added**
```
apiVersion: v1
items:
- apiVersion: migration.openshift.io/v1alpha1
  kind: MigPlan
  metadata:    creationTimestamp: 2019-06-10T21:07:42Z
    generation: 1
    name: migplan-sample
    namespace: mig
    resourceVersion: "23140"
  spec:
    destMigClusterRef:
      name: migcluster-aws
      namespace: mig
    migStorageRef:
      name: migstorage-sample
      namespace: mig
    namespaces:
    - mysql-persistent
    persistentVolumes:
    - name: pv5
      supportedActions:
      - copy
      - move
    srcMigClusterRef:
      name: migcluster-local
      namespace: mig
  status:
    conditions:
    - category: Error
      lastTransitionTime: 2019-06-10T21:42:49Z
      message: PV in `persistentVolumes` [pv5] has an unsupported `action`.
      reason: NotDone
      status: "True"
      type: PvInvalidAction
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:55Z
      message: The `persistentVolumes` list has been updated with discovered PVs.
      reason: Done
      status: "True"
      type: PvsDiscovered
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:57Z
      message: The storage resources have been created.
      status: "True"
      type: StorageEnsured
    - category: Required
      lastTransitionTime: 2019-06-10T21:42:59Z
      message: The migration registry resources have been created.
      status: "True"
      type: RegistriesEnsured
```

**Comment by djwhatle** [comment]
This is not resolved by https://github.com/fusor/mig-controller/pull/141 as it is today.

**Comment by djwhatle** [comment]
Resolved by #141
